File: leetcode/1062-remove-outermost-parentheses/RemoveOutermostParentheses.java
class Solution {
    public String removeOuterParentheses(String S) {
        StringBuilder result = new StringBuilder();
        
        int opened = 0;
        
        for (int i = 0; i < S.length(); i++) {
            char ch = S.charAt(i);
            
            if (ch == '(') {
                if (opened >= 1) {
                    result.append(ch);
                }
                
                opened++;
            } else {
                opened--;
                
                if (opened >= 1) {
                    result.append(ch);
                }
            }
        }
        
        return result.toString();
    }
}

File: leetcode/1021-remove-outermost-parentheses/remove-outermost-parentheses.js
/**
 * @param {string} S
 * @return {string}
 */
var removeOuterParentheses = function(S) {
  let level = 0;

  const result = [];

  for (const ch of S) {
    if (ch === '(') {
      if (level !== 0) {
        result.push(ch);
      }
      level++;
    } else if (ch === ')') {
      if (level !== 1) {
        result.push(ch);
      }
      level--;
    }
  }

  return result.join('')
};
